glusterd2/device/store-utils.go

```diff
@@ -0,0 +1,48 @@
+// Package device stores device information in the store
+package device
+
+import (
+	"encoding/json"
+
+	peer "github.com/gluster/glusterd2/glusterd2/peer"
+	"github.com/gluster/glusterd2/pkg/api"
+)
+
+const (
+	devicePrefix string = "/devices/"
+	peerPrefix   string = "peers/"
+)
+
+// GetDevice returns devices of specified peer from the store
+func GetDevice(peerID string) ([]api.DeviceInfo, error) {
+	peerInfo, err := peer.GetPeer(peerID)
```

Comment:
Instead, you can just set peer object in txn context in the handler, get it and edit it here.